sdks/java/io/singlestore/src/main/java/org/apache/beam/sdk/io/singlestore/SingleStoreIO.java

```diff
@@ -0,0 +1,955 @@
+ *     public KV<Integer, String> mapRow(ResultSet resultSet) throws Exception {
+ *       return KV.of(resultSet.getInt(1), resultSet.getString(2));
+ *     }
+ *   })
+ * );
+ * }</pre>
+ *
+ * <p>Query parameters can be configured using a user-provided {@link StatementPreparator}. For
+ * example:
+ *
+ * <pre>{@code
+ * pipeline.apply(SingleStoreIO.<KV<Integer, String>>read()
+ *   .withDataSourceConfiguration(DataSourceConfiguration.create("hostname:3306"))
+ *   .withQuery("select id,name from Person where name = ?")
+ *   .withStatementPreparator(new StatementPreparator() {
+ *     public void setParameters(PreparedStatement preparedStatement) throws Exception {
+ *       preparedStatement.setString(1, "Darwin");
+ *     }
+ *   })
+ *   .withRowMapper(new RowMapper<KV<Integer, String>>() {
+ *     public KV<Integer, String> mapRow(ResultSet resultSet) throws Exception {
+ *       return KV.of(resultSet.getInt(1), resultSet.getString(2));
+ *     }
+ *   })
+ * );
+ * }</pre>
+ *
+ * <h4>Parallel reading from a SingleStoreDB datasource</h4>
+ *
+ * <p>SingleStoreIO supports partitioned reading of all data from a table. To enable this, use
+ * {@link SingleStoreIO#readWithPartitions()}. This way of data reading is preferred because of the
+ * performance reasons.
+ *
+ * <p>The following example shows usage of {@link SingleStoreIO#readWithPartitions()}
+ *
+ * <pre>{@code
+ * pipeline.apply(SingleStoreIO.<Row>readWithPartitions()
+ *  .withDataSourceConfiguration(DataSourceConfiguration.create("hostname:3306")
+ *       .withUsername("username")
+ *       .withPassword("password"))
+ *  .withTable("Person")
+ *  .withRowMapper(new RowMapper<KV<Integer, String>>() {
+ *    public KV<Integer, String> mapRow(ResultSet resultSet) throws Exception {
+ *      return KV.of(resultSet.getInt(1), resultSet.getString(2));
+ *    }
+ *  })
+ * );
+ * }</pre>
+ *
+ * <h3>Writing to SingleStoreDB datasource</h3>
+ *
+ * <p>SingleStoreIO supports writing records into a database. It writes a {@link PCollection} to the
+ * database by converting data to CSV and sending it to the database with LOAD DATA query.
+ *
+ * <p>Like the source, to configure the sink, you have to provide a {@link DataSourceConfiguration}.
+ *
+ * <pre>{@code
+ * pipeline
+ *   .apply(...)
+ *   .apply(SingleStoreIO.<KV<Integer, String>>write()
+ *      .withDataSourceConfiguration(DataSourceConfiguration.create("hostname:3306")
+ *          .withUsername("username")
+ *          .withPassword("password"))
+ *      .withStatement("insert into Person values(?, ?)")
+ *      .withUserDataMapper(new UserDataMapper<KV<Integer, String>>() {
+ *        @Override
+ *        public List<String> mapRow(KV<Integer, String> element) {
+ *          List<String> res = new ArrayList<>();
+ *          res.add(element.getKey().toString());
+ *          res.add(element.getValue());
+ *          return res;
+ *        }
+ *      })
+ *    );
+ * }</pre>
+ */
+final class SingleStoreIO {
```

Comment:
Oops I believe this is a mistake. `final class SingleStoreUtil` and `public class SingleStoreUtil`